pkg/kubectl/cmd/version.go

```diff
@@ -46,36 +54,93 @@ func NewCmdVersion(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	}
 	cmd.Flags().BoolP("client", "c", false, "Client version only (no server required).")
 	cmd.Flags().BoolP("short", "", false, "Print just the version number.")
+	// default behavior is std.
+	cmd.Flags().String("output", "", "output format, options available are yaml and json")
 	cmd.Flags().MarkShorthandDeprecated("client", "please use --client instead.")
 	return cmd
 }
 
 func RunVersion(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
-	v := fmt.Sprintf("%#v", version.Get())
```

Comment:
I would order it like this:
1. get the client and (optionally) server versions
2. build the VersionObj struct with nil set appropriately
3. switch on cmdutil.GetFlagString(cmd, "output") (don't normalize their arg)
4. in the "" case, have all the existing output, including short output handling
5. in the json/yaml cases, marshal and output
6. in the default case, error